src/main/java/org/broadinstitute/hellbender/tools/spark/sv/NovelAdjacencyReferenceLocations.java

```diff
@@ -157,260 +60,20 @@ static EndConnectionType determineEndConnectionType(final ChimericAlignment chim
-                        final int d1 = (2*cn2 - cn1)*l + lambda;
-                        final int d2 = cn2*l + lambda;
-                        final double newErr = Math.abs(overlapOnHigherCNSequence-d1) + Math.abs(overlapOnLowerCNSequence-d2);
-                        if (newErr < err) {
-                            err = newErr;
-                            higherCnEst = cn1; lowerCnEst = cn2;
-                            unitLen= l; pseudoHomLen = lambda;
-                        }
-                        if (err<1)
-                            return new TandemRepeatComplication(lowerCnEst, higherCnEst, unitLen, pseudoHomLen);
-                    }
-                }
-            }
-        }
-
-        return new TandemRepeatComplication(lowerCnEst, higherCnEst, unitLen, pseudoHomLen);
-    }
-
-    private static final class TandemRepeatComplication {
-        final int lowerRepeatNumberEstimate;
-        final int higherRepeatNumberEstimate;
-        final int repeatedSeqLen;
-        final int pseduHomologyLen;
-
-        TandemRepeatComplication(final int lowerRepeatNumberEstimate,
-                                 final int higherRepeatNumberEstimate,
-                                 final int repeatedSeqLen,
-                                 final int pseduHomologyLen){
-            this.lowerRepeatNumberEstimate = lowerRepeatNumberEstimate;
-            this.higherRepeatNumberEstimate = higherRepeatNumberEstimate;
-            this.repeatedSeqLen = repeatedSeqLen;
-            this.pseduHomologyLen = pseduHomologyLen;
-        }
-    }
-
     // TODO: 12/11/16 again this does not deal with inter-chromosome translocation yet
     /**
      * Returns the reference coordinates of the left and right breakpoints implied by this chimeric alignment.
      * If there is homologous sequence represented in the alignments, it will be assigned to the side of the breakpoint
      * with higher reference coordinates.
      */
     @VisibleForTesting
-    static Tuple2<SimpleInterval, SimpleInterval> leftJustifyBreakpoints(final ChimericAlignment ca, final LocationComplication complication) {
+    static Tuple2<SimpleInterval, SimpleInterval> leftJustifyBreakpoints(final ChimericAlignment ca, final BreakpointComplications complication) {
 
         final int homologyLen = complication.homologyForwardStrandRep.length();
 
         final String leftBreakpointRefContig, rightBreakpointRefContig;
         final int leftBreakpointCoord, rightBreakpointCoord;
-        if (complication.dupSeqRepeatNumOnRef !=0 && complication.dupSeqRepeatNumOnCtg !=0) { // todo : development artifact-- assuming tandem duplication is not co-existing with inversion
-            final int dupLen = complication.dupSeqForwardStrandRep.length();
+        if (!complication.dupSeqRepeatUnitRefSpan.equals(BreakpointComplications.DUPSEQ_REPEAT_UNIT_NA_VALUE)) { // todo : development artifact-- assuming tandem duplication is not co-existing with inversion
```

Comment:
As I said earlier, it'd be nice to hide references to `BreakpointComplications.DUPSEQ_REPEAT_UNIT_NA_VALUE` behind a more straightforwardly named method.